subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -1667,6 +1668,14 @@ static int init_reset(void)
 	mem_init(conn_pool, sizeof(struct ll_conn),
 		 sizeof(conn_pool) / sizeof(struct ll_conn), &conn_free);
 
+	/* Invalidate connection handles, refer to ll_connected_get() */
+	for (uint16_t handle = 0U; handle < CONFIG_BT_MAX_CONN; handle++) {
+		struct ll_conn *conn;
+
+		conn = ll_conn_get(handle);
+		conn->lll.handle = LLL_HANDLE_INVALID;
```

Comment:
This seems to assume that the handles are already initialized to some value since you are calling `ll_conn_get` with handle 0, 1, 2, etc.

Reply:
`ll_conn_get()` translate to being array indexing of `conn_pool`.

https://github.com/zephyrproject-rtos/zephyr/blob/a1ba2249597b253b4eae0873dbef4677473a6fc2/subsys/bluetooth/controller/util/mem.c#L102

Reply:
Ah, I see. That's pretty misleading this case, as handle != index in many cases. 

If the `ll_conn_get` function at some point will be modified to not use handle as an index, then this will (silently?) break. 

There is one other place where handles are treat as indexes: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/ll_sw/ull_peripheral_iso.c#L67-L76

I think neither that, nor this, should treat handle as indexes, unless we can have some additional check in place to ensure this. 

Alternatively, we could define a `ll_conn_get_by_index` to take an index, rather than a handle.

Reply:
> If the ll_conn_get function at some point will be modified to not use handle as an index, then this will (silently?) break.

Currently, all `handle` generated/produced by the Controller are indexes. Any handle  provided by Host, example, advertising handle is converted to `handle` which are indexes internally.

There is no plan to not use indexes internally, to keep the handle to context de-referencing at O(1).

Reply:
Updated `ll_conn_get()` to check and return `NULL`. Added `LL_ASSERT` in callers; The commit has increased the full-feature code size by ~1.2 KB!.

These assertion checks are development checks on handle and connection context values that are verified at the first reference; I will try classify such assertion checks as conditionally compilable asserts under `CONFIG_ASSERT=y` so that a reduce code size Controller can be built for target testing (All assertion check be enabled for BabbleSIM CI testing).